### PR TITLE
build: update TypeScript to v4.2.4

### PR DESCRIPTION
**Pull request by Julien-Marcou**
Update TypeScript to be consistent across different packages.

This is a follow up for : https://github.com/angular/angular/pull/41618

## PR Checklist

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/compiler-cli/src/typescript_support.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import {compareVersions} from './diagnostics/typescript_version';
  * Note: this check is disabled in g3, search for
  * `angularCompilerOptions.disableTypeScriptVersionCheck` config param value in g3.
  */
-const MIN_TS_VERSION = '4.2.3';
+const MIN_TS_VERSION = '4.2.4';
```
Comment (alan-agius4) [inline packages/compiler-cli/src/typescript_support.ts:18]: 4.2.3, is supported and hence should be changed.

File: packages/compiler-cli/package.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
   },
   "peerDependencies": {
     "@angular/compiler": "0.0.0-PLACEHOLDER",
-    "typescript": ">=4.2.3 <4.3"
+    "typescript": ">=4.2.4 <4.3"
```
Comment (alan-agius4) [inline packages/compiler-cli/package.json:31]: Same as previous

File: packages/bazel/package.json [hunk 3]
```diff
@@ -29,7 +29,7 @@
     "@angular/compiler-cli": "0.0.0-PLACEHOLDER",
     "@bazel/typescript": ">=1.0.0",
     "terser": "^4.3.1",
-    "typescript": ">=4.2.3 <4.3",
+    "typescript": ">=4.2.4 <4.3",
```
Comment (alan-agius4) [inline packages/bazel/package.json:32]: Same, please revert.

**Review by alan-agius4 — COMMENTED** [review]
Apart from consistency I don’t see any benefit from doing this change.

That being said if we want consistence probably we should change TypeScript version to `~4.2.3`.

File: packages/compiler-cli/src/typescript_support.ts (hunk 1 above)
Comment (Julien-Marcou) [reply packages/compiler-cli/src/typescript_support.ts:18]: I was also wondering about this change ^^

Is there a reason we dropped the support for TypeScript 4.2.2 but should not drop 4.2.3 ?

Comment (alan-agius4) [reply packages/compiler-cli/src/typescript_support.ts:18]: We never added support for TS 4.2.2

**Comment by Julien-Marcou** [comment]
> Apart from consistency I don’t see any benefit from doing this change.
> 
> That being said if we want consistence probably we should change TypeScript version to `~4.2.3`.

Yes, I guess you're right, but TypeScript 4.2.3 has a big issue with `@deprecated` doc tag, and I don't see why people would want 4.2.3 over 4.2.4.

Even though there is not much benefit, I think it still makes it easier to track down the version of TypeScript we are using and so makes it easier to upgrade in the future (it also allows for the package manager to download only one version of TypeScript and caching it, making the install a tiny bit faster ^^)

**Comment by alan-agius4** [comment]
> Yes, I guess you're right, but TypeScript 4.2.3 has a big issue with @deprecated doc tag, and I don't see why people would want 4.2.3 over 4.2.4.

Note:  `~4.2.3` will still match and download `4.2.4` (Unless there is a lock file pinning this), also the TypeScript version here is not the one used to develop Angular applications. It's the version used to develop the Angular FW itself.

None the less I am okay with this change.

**Review by alan-agius4 — APPROVED** [review]

**Review by josephperrott — APPROVED** [review]
LGTM

As noted, these are not versions of typescript expected for any downstream usage of angular.

**Review by josephperrott — APPROVED** [review]
Reviewed-for: global-approvers

**Pull request closed without merging**